server/controllers/payroll/reports/payslipGenerator.js

```diff
@@ -0,0 +1,230 @@
+
+  getHolidays.forEach(item => {
+    transactionHoliday.addQuery(item.query, item.params);
+  });
+
+  getOffDays.forEach(item => {
+    transactionOffDay.addQuery(item.query, item.params);
+  });
+
+  transaction.execute()
+    .then((results) => {
+      employeeData.forEach(emp => {
+        emp.employee.rubricTaxable = [];
+        emp.employee.rubricNonTaxable = [];
+        emp.employee.rubricDiscount = [];
+        let somRubTaxable = 0;
+        let somRubNonTaxable = 0;
+        let somChargeEmployee = 0;
+        let somChargeEnterprise = 0;
+
+        results.forEach(rubEmployee => {
+          rubEmployee.forEach(rubrics => {
+            if (emp.employee.employee_uuid === rubrics.employee_uuid) {
+              rubrics.ratePercentage = rubrics.is_percent ? rubrics.value : '---';
+
+              // Get Rubric Taxable
+              if (!rubrics.is_discount && !rubrics.is_social_care) {
+                somRubTaxable += rubrics.result;
+                emp.employee.rubricTaxable.push(rubrics);
+              }
+
+              // Get Rubric Non Taxable
+              if (!rubrics.is_discount && rubrics.is_social_care) {
+                somRubNonTaxable += rubrics.result;
+                emp.employee.rubricNonTaxable.push(rubrics);
+              }
+
+              // Get Charge
+              if (rubrics.is_discount) {
+                if (rubrics.is_employee) {
+                  rubrics.chargeEmployee = rubrics.result;
+                  rubrics.chargeEnterprise = 0;
+
+                  somChargeEmployee += rubrics.result;
+                } else {
+                  rubrics.chargeEmployee = 0;
+                  rubrics.chargeEnterprise = rubrics.result;
+
+                  somChargeEnterprise += rubrics.result;
+                }
+
+                emp.employee.rubricDiscount.push(rubrics);
+              }
+            }
+          });
+        });
+
+        emp.employee.somRubTaxable = somRubTaxable;
+        emp.employee.somRubNonTaxable = somRubNonTaxable;
+        emp.employee.somChargeEmployee = somChargeEmployee;
+        emp.employee.somChargeEnterprise = somChargeEnterprise;
+      });
+
+      return transactionHoliday.execute();
+    })
+    .then(results => {
+      employeeData.forEach(emp => {
+        emp.employee.holidaysPaid = [];
+        emp.employee.dailyWorkedValue = emp.employee.daily_salary * emp.employee.working_day;
+
+        results.forEach(holidayEmployee => {
+          holidayEmployee.forEach(holiday => {
+            if (emp.employee.uuid === holiday.paiement_uuid) {
+              holiday.dailyRate = holiday.value / holiday.holiday_nbdays;
+              emp.employee.holidaysPaid.push(holiday);
+            }
+          });
+        });
+      });
+
+      return transactionOffDay.execute();
+    })
+    .then(results => {
+      employeeData.forEach(emp => {
+        emp.employee.offDaysPaid = [];
+        results.forEach(offDayEmployee => {
+          offDayEmployee.forEach(offDay => {
```

Comment:
This is a good idea but will not solve the problem because by using a filter we will add in the array 'emp.employee.offDaysPaid' the filtered offDayEmployee while we only need to progressively add the elements offDayEmployee which respects the condition

Reply:
It won't produce the same result?  I thought it would. :(